lib/hooks.php

```diff
@@ -32,15 +32,55 @@ class Hooks {
 	 * @param array $params
 	 */
 	public static function postLogin($params) {
+		$request = \OC::$server->getRequest();
+		$authorizationHeader = $request->server['HTTP_AUTHORIZATION'];
+		$userAgent = $request->server['USER_AGENT'];
+		$split = explode(' ', $authorizationHeader);
+		if(count($split) === 2) {
+			$decoded = base64_decode($split[1]);
+			if($decoded !== false) {
+				$splitDecoded = explode(':', $decoded);
+				if (count($splitDecoded) === 2) {
+					$authorizationHeader = $splitDecoded[0];
+				} else {
+					$authorizationHeader = 'splitting decoded header on ":" failed';
+				}
+			} else {
+				$authorizationHeader = 'base64 decode failed';
+			}
+		} else {
+			$authorizationHeader = 'splitting header on " " failed';
+		}
+		$info = "AuthUser:$authorizationHeader,UserAgent:$userAgent";
+
+
 		$dbConnection = \OC::$server->getDatabaseConnection();
-		$query = $dbConnection->prepare('INSERT INTO `*PREFIX*login_sessions` (`user_id`, `session_id`, `timestamp`) VALUES(?, ?, ?)');
+		$query = $dbConnection->prepare('INSERT INTO `*PREFIX*login_sessions` (`user_id`, `session_id`, `timestamp`, `info`) VALUES(?, ?, ?, ?)');
 		$query->bindValue(1, $params['uid']);
 		$query->bindValue(2, session_id());
 		$query->bindValue(3, time());
+		if(strlen($info) > 255) {
+			$info = substr($info, 0, 252) . '...';
+		}
+		$query->bindValue(4, $info);
 		try {
 			$query->execute();
 		} catch (\Exception $e) {
-			\OC::$server->getLogger()->critical($e->getMessage(), array('app' => 'session_login_tracker'));
+			$logger = \OC::$server->getLogger();
+			$logger->critical($e->getMessage(), array('app' => 'session_login_tracker'));
+			$logger->critical($info, array('app' => 'session_login_tracker'));
+			$sessionId = session_id();
+			if (function_exists('memcache_connect')) {
+				$sessionKey = 'memc.sess.key.' . $sessionId;
+				$memcache = memcache_connect('memcached.example.org', 11211);
```

Comment:
Just for the record for all others: This needs to get adjusted based on the environment. :)